src/jit/codegenarm64.cpp

```diff
@@ -9396,6 +9388,38 @@ void                CodeGen::genArm64EmitterUnitTests()
 }
 #endif // defined(DEBUG)
 
+
+//------------------------------------------------------------------------------------------------ //
+// getFirstArgWithStackSlot - returns the first argument with stack slot on the caller's frame.
+//
+// Return value:
+//    The number of the first argument with stack slot on the caller's frame.
+
+unsigned
+CodeGen::getFirstArgWithStackSlot()
+{
+    // Iterate over all the local variables in the lclvartable.
+    // They contain all the implicit argumets - thisPtr, retBuf,
+    // generic context, PInvoke cookie, var arg cookie,no-standard args, etc.
+    unsigned baseVarNum = 0;
+    LclVarDsc* varDsc = nullptr;
+    for (unsigned i = 0; i < compiler->lvaCount; i++)
```

Comment:
lvaCount would imply that we would iterate over all lcl vars both params and locals.  To just iterate only over arguments/params use compiler->info.compArgsCount

During testing didn't the assert on line 9411 fire?